doc/ssl-performance.md

```diff
@@ -0,0 +1,30 @@
+# SSL in gRPC and performance
+
+The SSL requirement of gRPC isn't necessarily making it easy to integrate. The HTTP/2 protocol requires ALPN support, which is a fairly new handshake protocol only supported by recent implementations.
+
+As a result, we've tried hard to provide a smooth experience to our users when compiling and distributing gRPC, but this may come at performance costs due to this. More specifically, we will sometime build the SSL library by disabling assembly code, which can impact performances by an order of magnitude when processing encrypted streams.
+
+Build system | Condition | Platform | Assembly code
+---|---|---|--
+Makefile | with OpenSSL 1.0.2 development files | all | :heavy_check_mark:
+Makefile | all other cases | all | :x:
+Bazel | | Linux | :heavy_check_mark:
+Bazel | | MacOS | :heavy_check_mark:
+Bazel | | Windows | :x:
+CMake | | Windows | :x:
+CMake | | all others | :heavy_check_mark:
```

Comment:
actually, CMake's story is a bit more complicated.

- When building from submodule (the default), we set OPENSSL_NO_ASM for all platforms, not just for Windows:
https://github.com/grpc/grpc/blob/e916e79cb88b007559e079a8cac2250105a76954/cmake/ssl.cmake#L20 . This is not necessary on Linux and Mac, but it's necessary on Windows if one wants to be able to build from generated visual studio projects. If one uses ninja on Win instead of VS, building with assembly optimizations is possible.

- If we not building from submodule, use the system-installed openssl (which is assumed to have assembly optimizations enabled).

Reply:
Ah, right, I was confused by the comment, okay.